crates/server/src/store/forwarder/authority.rs

```diff
@@ -116,11 +117,15 @@ impl Authority for ForwardAuthority {
 
     async fn search(
         &self,
-        query: &LowerQuery,
+        request_info: RequestInfo<'_>,
         lookup_options: LookupOptions,
     ) -> Result<Self::Lookup, LookupError> {
-        self.lookup(query.name(), query.query_type(), lookup_options)
-            .await
+        self.lookup(
+            request_info.query.name(),
```

Comment:
Why not just send the `RequestInfo` in here?

Reply:
Lookup is used internally by most of the authorities for getting records, for things like SOA. It would be a bigger change to add to that interface. (Really, that method shouldn’t be public).

I did consider it.

Reply:
In essence, I think think for things like authorizing the request, or switch records based on src addr, that would happen in "search", then lookup performs all the actions to collect the records. That's the split between these two interfaces today.